src/st/no3/377D4.c

```diff
@@ -1001,7 +1144,7 @@ void EntityMermanRockRightSide(Entity* self) {
             tilePos += 0x30;
         }
 
-        if (D_8003BDEC[51] & 2) { /* 0 0 0 0 0 0 1 0 = Half broken */
+        if (D_8003BDEC[jewelSwordRoomUnlock] & rockBroken) {
```

Comment:
A lot of changes unrelated to decompiling EntityStairwayPiece are showing up in this PR, this is probably going to slow down PR review. Could you make a second PR for the other stuff?

Reply:
Just checked, pretty sure you can read through it in one minute, I only documented merman rock left and right and included two duplicates, it isn't that much.